### DISCUSSION TITLE
Chart JS

### DISCUSSION THREAD

**Discussion by mabdalla123** [abpframework/abp#19154] [Q&A]
How can i use ChartJs with MVC Application , there is no clear way to do it

**Comment by realLiangshiwei** [comment]
ABP is a standard ASPNETCore application, 

You can easily get the answer on Google: https://www.google.com/search?q=How+to+use+ChartJS+in+ASP+NET+Core

**Reply by mabdalla123** [reply]
What I meant is installing it .
When I tried to install it following the documentation , installing client side library IAM getting an error, a js error so I figured I must have missed something
For now IAM using CDN link but I would like to install it

**Comment by ajmalfaris11** [comment]
Integrating Chart.js with an MVC (Model-View-Controller) application can be done in several ways, depending on the specific requirements and technologies you're using. Below, I'll outline a general approach to integrating Chart.js with an MVC application:

1. **Install Chart.js**: Start by including Chart.js in your project. You can either download it and include it directly in your project or install it via a package manager like npm or yarn. For example, if you're using npm, you can install Chart.js by running:
   
   ```
   npm install chart.js
   ```

2. **Add Chart.js to Your View**: In your MVC view where you want to display the chart, include the Chart.js library. You can add it either directly in the HTML file or use a script bundler if you're using one in your MVC application.

   ```html
   <!-- Include Chart.js library -->
   <script src="path/to/chart.js"></script>
   ```

3. **Create a Canvas Element**: Chart.js renders charts onto HTML5 canvas elements. Create a canvas element in your view where you want the chart to appear.

   ```html
   <canvas id="myChart"></canvas>
   ```

4. **Write JavaScript to Render the Chart**: Write JavaScript code to render the chart using Chart.js. This code typically goes in a `<script>` tag in your view or in a separate JavaScript file that is included in your view.

   ```javascript
   // JavaScript code to render the chart
   var ctx = document.getElementById('myChart').getContext('2d');
   var myChart = new Chart(ctx, {
       type: 'bar',
       data: {
           labels: ['Red', 'Blue', 'Yellow', 'Green', 'Purple', 'Orange'],
           datasets: [{
               label: '# of Votes',
               data: [12, 19, 3, 5, 2, 3],
               backgroundColor: [
                   'rgba(255, 99, 132, 0.2)',
                   'rgba(54, 162, 235, 0.2)',
                   'rgba(255, 206, 86, 0.2)',
                   'rgba(75, 192, 192, 0.2)',
                   'rgba(153, 102, 255, 0.2)',
                   'rgba(255, 159, 64, 0.2)'
               ],
               borderColor: [
                   'rgba(255, 99, 132, 1)',
                   'rgba(54, 162, 235, 1)',
                   'rgba(255, 206, 86, 1)',
                   'rgba(75, 192, 192, 1)',
                   'rgba(153, 102, 255, 1)',
                   'rgba(255, 159, 64, 1)'
               ],
               borderWidth: 1
           }]
       },
       options: {
           scales: {
               y: {
                   beginAtZero: true
               }
           }
       }
   });
   ```

5. **Populate Data Dynamically (Optional)**: If your chart data needs to be dynamically populated based on server-side data, you can use Ajax or any other method to fetch the data and then update the chart accordingly.

This is a basic example of how to integrate Chart.js with an MVC application. Depending on your specific requirements and the MVC framework you're using, you may need to adapt this approach accordingly.
